src/lambda/commands/deploySamApplication.ts

```diff
@@ -116,23 +116,22 @@ export async function deploySamApplication(
 
         // successful deploy: retain S3 bucket for quick future access
         // JSON object of type: { [profile: string]: { [region: string]: bucket } }
-        const bucketsJson = settings.readSetting<string | undefined>(CHOSEN_BUCKET_KEY)
+        const bucketsJson = settings.readSetting<SavedBuckets | undefined>(CHOSEN_BUCKET_KEY)
         const profile = awsContext.getCredentialProfileName()
         if (profile) {
             try {
-                const buckets = bucketsJson ? JSON.parse(bucketsJson) : undefined
                 if (!bucketsJson) {
                     writeFreshJson(profile, deployWizardResponse, settings)
                 } else {
                     settings.writeSetting(
                         CHOSEN_BUCKET_KEY,
-                        JSON.stringify({
-                            ...buckets,
+                        {
+                            ...bucketsJson,
                             [profile]: {
-                                ...(buckets[profile] ? buckets[profile] : {}),
+                                ...(bucketsJson[profile] ? bucketsJson[profile] : {}),
```

Comment:
We should probably move this logic into `writeFreshJson()` (and rename it to `save()` or something)